recipes/management/commands/seed_full.py

```diff
@@ -0,0 +1,122 @@
+import csv
+from importlib.resources import files
+
+from django.core.management.base import BaseCommand
+
+from recipes.models import Ingredient, Recipe, Step, StepIngredient, Tag, Unit
+
+
+class Command(BaseCommand):
+    help = "Seed database with sample data from CSV. WARNING: deletes existing data."
+
+    def handle(self, *args, **kwargs):
+        self.stdout.write("Seeding database...")
+
+        # clear old data
+        Recipe.objects.all().delete()
+        Ingredient.objects.all().delete()
+        Unit.objects.all().delete()
+        Tag.objects.all().delete()
+        Step.objects.all().delete()
+        StepIngredient.objects.all().delete()
+
+        # create tags
+        tag_slugs = [
+            "main-dish",
+            "side-item",
+            "dessert",
+            "snack",
+            "veggie",
+            "chicken",
+            "dressing",
+            "bread",
+            "chutney",
+        ]
+        Tag.objects.bulk_create([Tag(slug=slug) for slug in tag_slugs])
+
+        # create units
+        unit_content = [
+            ("unit", "units", "", ""),
+            ("teaspoon", "teaspoons", "tsp", "tsp"),
+            ("tablespoon", "tablespoons", "tbsp", "tbsp"),
+            ("ounce", "ounces", "oz", "oz"),
+            ("cup", "cups", "", ""),
+            ("pound", "pounds", "lb", "lbs"),
+            ("bunch", "bunches", "", ""),
+            ("dash", "dashes", "", ""),
+            ("leaf", "leaves", "", ""),
+        ]
+        Unit.objects.bulk_create(
+            [
+                Unit(
+                    name=name,
+                    name_plural=name_plural,
+                    abbr_singular=abbr_singular,
+                    abbr_plural=abbr_plural,
+                )
+                for name, name_plural, abbr_singular, abbr_plural in unit_content
+            ]
+        )
+        unit_mapper = {
+            "unit": "unit",
+            "tbsp": "tablespoon",
+            "tsp": "teaspoon",
+            "oz": "ounce",
+            "c": "cup",
+            "leaves": "leaf",
+        }
+
+        # parse csv file
+        data_path = files("recipes.data").joinpath("seed_data.csv")
```

Comment:
wow TIL, did not know about files in importlib, nice

Reply:
cool, ok I need to read up on this!

<img width="876" height="374" alt="image" src="https://github.com/user-attachments/assets/5e2b6ba5-c663-4b39-b813-2c87f71433a8" />

<img width="871" height="917" alt="image" src="https://github.com/user-attachments/assets/470dfd79-224c-4d3f-9f03-1331fdec9678" />